File: VB/MvcCustomController/App_Start/DashboardConfig.vb
Imports System.Web.Hosting
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWeb
Imports DevExpress.DashboardWeb.Mvc
Imports DevExpress.DataAccess.Excel
Imports DevExpress.DataAccess.Sql

Public Class DashboardConfig
    Public Shared Sub RegisterService(routes As RouteCollection)
		routes.MapDashboardRoute("dashboardControl", "DefaultDashboard")

		Dim dashboardFileStorage As New DashboardFileStorage("~/App_Data/Dashboards")
		DashboardConfigurator.Default.SetDashboardStorage(dashboardFileStorage)

		' Uncomment this string to allow end users to create new data sources based on predefined connection strings.
		'DashboardConfigurator.Default.SetConnectionStringsProvider(new DevExpress.DataAccess.Web.ConfigFileConnectionStringsProvider());

		Dim dataSourceStorage As New DataSourceInMemoryStorage()

		' Registers an SQL data source.
		Dim sqlDataSource As New DashboardSqlDataSource("SQL Data Source", "NWindConnectionString")
		Dim query As SelectQuery = SelectQueryFluentBuilder.AddTable("SalesPerson").SelectAllColumns().Build("Sales Person")
		sqlDataSource.Queries.Add(query)
		dataSourceStorage.RegisterDataSource("sqlDataSource", sqlDataSource.SaveToXml())

		' Registers an Excel data source.
		Dim excelDataSource As New DashboardExcelDataSource("Excel Data Source")
		excelDataSource.FileName = HostingEnvironment.MapPath("~/App_Data/Sales.xlsx")
		excelDataSource.SourceOptions = New ExcelSourceOptions(New ExcelWorksheetSettings("Sheet1"))
		dataSourceStorage.RegisterDataSource("excelDataSource", excelDataSource.SaveToXml())

		DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage)
	End Sub
End Class

File: CS/MvcCustomController/App_Start/DashboardConfig.cs
using System.Web.Hosting;
using System.Web.Routing;
using DevExpress.DashboardCommon;
using DevExpress.DashboardWeb;
using DevExpress.DashboardWeb.Mvc;
using DevExpress.DataAccess.Excel;
using DevExpress.DataAccess.Sql;

namespace MvcCustomController {
    public class DashboardConfig {
        public static void RegisterService(RouteCollection routes) {
            routes.MapDashboardRoute("dashboardControl", "DefaultDashboard");

            DashboardFileStorage dashboardFileStorage = new DashboardFileStorage("~/App_Data/Dashboards");
            DashboardConfigurator.Default.SetDashboardStorage(dashboardFileStorage);

            // Uncomment this string to allow end users to create new data sources based on predefined connection strings.
            //DashboardConfigurator.Default.SetConnectionStringsProvider(new DevExpress.DataAccess.Web.ConfigFileConnectionStringsProvider());

            DataSourceInMemoryStorage dataSourceStorage = new DataSourceInMemoryStorage();

            // Registers an SQL data source.
            DashboardSqlDataSource sqlDataSource = new DashboardSqlDataSource("SQL Data Source", "NWindConnectionString");
            SelectQuery query = SelectQueryFluentBuilder
                .AddTable("SalesPerson")
                .SelectAllColumns()
                .Build("Sales Person");
            sqlDataSource.Queries.Add(query);
            dataSourceStorage.RegisterDataSource("sqlDataSource", sqlDataSource.SaveToXml());

            // Registers an Excel data source.
            DashboardExcelDataSource excelDataSource = new DashboardExcelDataSource("Excel Data Source");
            excelDataSource.FileName = HostingEnvironment.MapPath(@"~/App_Data/Sales.xlsx");
            excelDataSource.SourceOptions = new ExcelSourceOptions(new ExcelWorksheetSettings("Sheet1"));
            dataSourceStorage.RegisterDataSource("excelDataSource", excelDataSource.SaveToXml());

            DashboardConfigurator.Default.SetDataSourceStorage(dataSourceStorage);         
        }
    }
}